lib/osutil/atomic.go

```diff
@@ -50,6 +51,24 @@ func CreateAtomic(path string) (*AtomicWriter, error) {
 	return w, nil
 }
 
+// AppendAtomic is like CreateAtomic, but in case a file already exists, it
+// copies its contents to the temporary file. In case of an error on copying,
+// it proceeds as if there wasn't any file to append to.
+func AppendAtomic(path string) (*AtomicWriter, error) {
+	w, err := CreateAtomic(path)
+	if err != nil {
+		return nil, err
+	}
+
+	orig, err := os.Open(path)
+	if err == nil {
+		io.Copy(w, orig)
+	}
+	orig.Close()
```

Comment:
This is probably a nil pointer crash waiting to happen when the open fails; the close should be inside the if.